production/bin/create_configs.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# generate configurartion files with group and world read permissions.
+umask 022
+
+echo "Starting application configuration. Processing ENV settings."
+
+# Move default configs into place.
+# For most environments, these settings are overridden in ENV set from SSM.
+(
+  cd config
+  if [ ! -f config.yml ]; then
+    cp config.yml.example config.yml
+  fi
+  if [ ! -f database.yml ]; then
+    cp database.yml.example database.yml
+  fi
+  if [ ! -f sidekiq.yml ]; then
+    cp sidekiq.yml.example sidekiq.yml
+  fi
+  if [ ! -f cookies.txt ]; then
+    # Copy a default cookie file into place, even if using S3 storage for cookies.
+    cp cookies.txt.example config/cookies.txt
+  fi
+
+  # Generate sidekiq config from SSM:
+  WORKTMP=$(mktemp)
+  if [[ -z ${sidekiq_config+x} ]]; then
+    echo "Error: missing sidekiq_config ENV setting. Using defaults."
+  else
+    echo ${sidekiq_config} | base64 -d > $WORKTMP
+    if (( $? != 0 )); then
+      echo "Error: could not decode ENV var: ${sidekiq_config} . Using defaults."
+      rm $WORKTMP
+    else
+      echo "Using decoded sidekiq config from ENV var: ${sidekiq_config} ."
+      mv $WORKTMP sidekiq.yml
+      sha1sum sidekiq.yml
+    fi
+  fi
+
+  # Generate database configuration from SSM:
+  WORKTMP=$(mktemp)
+  if [[ -z ${database_config+x} ]]; then
+    echo "Error: missing database_config ENV setting. Using defaults."
+  else
+    echo ${database_config} | base64 -d > $WORKTMP
+    if (( $? != 0 )); then
+      echo "Error: could not decode ENV var: ${database_config} . Using defaults."
+      rm $WORKTMP
+    else
+      echo "Using decoded database config from ENV var: ${database_config} ."
+      mv $WORKTMP database.yml
+      sha1sum database.yml
+    fi
+  fi
+
+  # Populate production environment config from SSM:
+  WORKTMP=$(mktemp)
+  if [[ -z ${environments_production+x} ]]; then
```

Comment:
Could you help me understand how this works? I want to make sure that these values override the default ones given in config.yml, since right now with the file copied above those would all be development values. Where do variables like `environments_production` come from?

Reply:
I realized that a confluence document describing the SSM driven configuration of Check applications would be useful here. Let me get this posted and point at the relevant parts in the doc. (Basically, the SSM parameter of this name, is mapped into the task ENV with same name, and this base64 value is used to generate the config file...)

Reply:
Super helpful, thanks!